torchx/util/role_to_pod.py

```diff
@@ -0,0 +1,183 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+from torchx.specs.api import (
+    BindMount,
+    DeviceMount,
+    Role,
+    VolumeMount,
+)
+from typing import Dict, Optional
+from kubernetes.client.models import V1Pod
```

Comment:
can we wrap this in a TYPE_CHECKING block? we late import kubernetes to avoid adding long import times. 

Though we do have other changes that makes this a bit less necessary